docs/docs/resources/architecture/components-hierarchy.md

```diff
@@ -0,0 +1,18 @@
+<!-- I dislike how the mermaid diagram looks, hence I drew it in draw.io. ~Ivan -->
+
+<figure markdown>
+  ![kpops-component-hierarchy](/images/kpops-components_class-diagram.png)
+  <figcaption>KPOps component hierarchy</figcaption>
+</figure>
+
+<!-- ```mermaid
```

Comment:
why did you replace this diagram? does it not work with mkdocs?

Reply:
ah, sorry I didn't see your comment there. whatever works best :)

Reply:
Mermaid is much better in terms of scalability and stylistic consistency of the docs, as changing the diagram is very easy and we can use global configs. I'm gonna use mermaid. (Although I still don't really like how it looks.)

Reply:
re-opening this because I missed the discussion about why we changed this again? 😄

Reply:
@sujuka99 your last comment says we should stick with mermaid, however it is commented out

Reply:
Ah yes after I resolved this one, @raminqaf commented that he doesn't like the empty rectangles in the components, so since you didn't indicate a strong preference and I do agree with @raminqaf that the mermaid-generated one doesn't look good, I reverted to using the hand-drawn one.

The mermaid code is still there, commented out to remind the alternative and to provide a starting point for when inevitably drawing diagrams by hand is not viable anymore.

Reply:
thanks for the explanation